caffe2/operators/utility_ops.h

```diff
@@ -738,6 +738,122 @@ class ScatterAssignOp : public Operator<Context> {
   INPUT_TAGS(DATA, INDICES, SLICES);
 };
 
+template <class Context>
```

Comment:
I'd suggest to make this op in a separate cc and h file. It is not a very good design to put so many ops in utility_ops.